base/ca/src/main/java/org/dogtagpki/server/ca/v2/CertRequestServlet.java

```diff
@@ -0,0 +1,450 @@
+import javax.servlet.http.HttpSession;
+
+import org.dogtagpki.server.ca.CAEngine;
+import org.dogtagpki.server.ca.CAServlet;
+import org.dogtagpki.server.ca.rest.ProfileService;
+import org.mozilla.jss.netscape.security.x509.X500Name;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netscape.ca.CertificateAuthority;
+import com.netscape.certsrv.authentication.AuthCredentials;
+import com.netscape.certsrv.authentication.EAuthException;
+import com.netscape.certsrv.authorization.EAuthzException;
+import com.netscape.certsrv.base.BadRequestDataException;
+import com.netscape.certsrv.base.BadRequestException;
+import com.netscape.certsrv.base.ConflictingOperationException;
+import com.netscape.certsrv.base.EBaseException;
+import com.netscape.certsrv.base.PKIException;
+import com.netscape.certsrv.base.ResourceNotFoundException;
+import com.netscape.certsrv.base.UnauthorizedException;
+import com.netscape.certsrv.ca.AuthorityID;
+import com.netscape.certsrv.cert.CertEnrollmentRequest;
+import com.netscape.certsrv.cert.CertRequestInfo;
+import com.netscape.certsrv.cert.CertRequestInfos;
+import com.netscape.certsrv.profile.ProfileAttribute;
+import com.netscape.certsrv.profile.ProfileDataInfo;
+import com.netscape.certsrv.profile.ProfileDataInfos;
+import com.netscape.certsrv.profile.ProfileInput;
+import com.netscape.certsrv.request.RequestId;
+import com.netscape.certsrv.request.RequestNotFoundException;
+import com.netscape.certsrv.util.JSONSerializer;
+import com.netscape.cms.profile.common.Profile;
+import com.netscape.cms.servlet.cert.CertRequestInfoFactory;
+import com.netscape.cms.servlet.cert.EnrollmentProcessor;
+import com.netscape.cms.servlet.cert.RenewalProcessor;
+import com.netscape.cms.servlet.processors.CAProcessor;
+import com.netscape.cmscore.profile.ProfileSubsystem;
+import com.netscape.cmscore.request.Request;
+import com.netscape.cmscore.request.RequestRepository;
+/**
+ * @author Marco Fargetta {@literal <mfargett@redhat.com>}
+ */
+@WebServlet(
+        name = "caCertRequest",
+        urlPatterns = "/v2/certrequests/*")
+public class CertRequestServlet extends CAServlet {
+
+    private static final long serialVersionUID = 1L;
+    private static Logger logger = LoggerFactory.getLogger(CertRequestServlet.class);
+
+    @Override
+    public void get(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        HttpSession session = request.getSession();
+        logger.debug("CertRequestServlet.get(): session: {}", session.getId());
+
+        response.setContentType("application/json");
+        PrintWriter out = response.getWriter();
+        if(request.getPathInfo() == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, request.getRequestURI());
+            return;
+        }
+
+        String[] pathElement = request.getPathInfo().substring(1).split("/");
+        if (pathElement.length == 1) {
+            if (pathElement[0].equals("profiles")) {
```

Comment:
I've updated these pages so we can see how the API looks like (feel free to update these pages too):
https://github.com/dogtagpki/pki/wiki/CA-REST-API-v1
https://github.com/dogtagpki/pki/wiki/CA-REST-API-v2

I think we can change `/ca/v2/certrequests/profiles` to `/ca/v2/certrequest-templates` or something like that so we don't have to treat `"profiles"` as a special keyword to distinguish it from `/ca/v2/certrequests/{id}`. That way we can also move template-related code into a separate servlet.

Reply:
Your proposed API change seems reasonable, it makes  sense to move the profiles from the certrequests endpoint to a separate one. I was thinking to reorganise the API after re-implementing all v1 endpoint because we can test more easily the commands between the two versions and be sure to have everything working properly before to reorganise and publish them.